### PR TITLE
OSD-11934 Default Ingress Timeout

### PR DESCRIPTION
**Pull request by luis-falcon**
For the regular app ingress we override the default timeout to 30 minutes [1] however we do not do this for CDO - we need to have a consistent setting on all our default ingress settings.

For this PR I am creating the variable that holds the duration for the ELB idle time out. I create the util functions that help us get the the infrastructure in ordre to identify if we are on AWS cluster (if yes we set the time to the new idle time out).

I also added the necessary permissions on the clusterRoles for the operator to be able to query the infrastructure resource on - config.openshift.io. I wil creating a card in a bit requesting unit testing for the util functions.

### CODE REVIEW

**Comment by luis-falcon** [comment]
For the regular app ingress we override the default timeout to 30 minutes [1] however we do not do this for CDO - we need to have a consistent setting on all our default ingress settings.

For this PR I am creating the variable that holds the duration for the ELB idle time out. I create the util functions that help us get the the infrastructure in ordre to identify if we are on AWS cluster (if yes we set the time to the new idle time out). 

I also added the necessary permissions on the clusterRoles for the operator to be able to query the infrastructure resource on   - config.openshift.io.  I wil creating a card in a bit requesting unit testing for the util functions.

**Comment by fahlmant** [comment]
@luis-falcon Can you move your comment to the description of the PR?

**Review by rafael-azevedo — APPROVED** [review]

File: controller/customdomain_controller_test.go [hunk 1]
```diff
@@ -642,3 +654,27 @@ func TestCustomDomainController(t *testing.T) {
 		t.Fatalf("ingresscontroller %s was not deleted!", instanceName)
 	}
 }
+
+func UpdateCloudPlatformToAWS(kclient client.Client, cloudPlatformName string) error {
```
Comment (dustman9000) [inline controller/customdomain_controller_test.go:658]: This function should be renamed to UpdatePlatformStatus and only take one argument: `"kclient client.Client"`. The configv1 CR always called "cluster".
Can you also add a comment above this function to describe it's purpose?

Comment (luis-falcon) [reply controller/customdomain_controller_test.go]: Done.

**Review by luis-falcon — COMMENTED** [review]
Changes to function name done.

File: controller/customdomain_controller_test.go [hunk 2]
```diff
@@ -253,6 +253,14 @@ func TestCustomDomainController(t *testing.T) {
 		objs = append(objs, cd)
 	}
 
+	infra := &configv1.Infrastructure{
```
Comment (dustman9000) [inline controller/customdomain_controller_test.go:256]: It doesn't look like you are testing anything new in this function. After running the reconcile, the timeout should be the new default timeout, but it doesn't look like you are testing that at all in this function.

File: deploy/crds/managed.openshift.io_customdomains.yaml [hunk 3]
```diff
@@ -49,11 +49,11 @@ spec:
                 description: Certificate points to the custom TLS secret
                 properties:
                   name:
-                    description: Name is unique within a namespace to reference a
+                    description: name is unique within a namespace to reference a
```
Comment (dustman9000) [inline deploy/crds/managed.openshift.io_customdomains.yaml:52]: Why was this changed?

File: controller/customdomain_utils.go [hunk 4]
```diff
@@ -228,3 +233,39 @@ func randSeq(n int) string {
 	}
 	return string(b)
 }
+
+// GetPlatformType returns the cloud platform type for the cluster
+func GetPlatformType(kclient client.Client) (*configv1.PlatformType, error) {
```
Comment (dustman9000) [inline controller/customdomain_utils.go:238]: It looks like these were just copied from https://github.com/openshift/cloud-ingress-operator/blob/master/pkg/utils/infrastructure.go
Why don't you just import "github.com/cloud-ingress-operator/pkg/utils" instead?

Comment (luis-falcon) [reply controller/customdomain_utils.go:238]: Hey Dustin. I was doing this for some operators but at some I believe Christoph or Taylor, I don't remember which one told me is not that good of a practice importing one operator into another (I might dreaming on this but I vaguely remember having this conversation with one of the leads). This is why Instead of importing it I just got the full code. But this import was my first instinct also, so I do not mind making this change if it is a must for you, Let me know.

File: deploy/crds/managed.openshift.io_customdomains.yaml (hunk 3 above)
Comment (luis-falcon) [reply deploy/crds/managed.openshift.io_customdomains.yaml:52]: To be honest I do not know. This is something that was breaking on the prow validate and  @sam-nguyen7 privided me with this command `./boilerplate/_lib/container-make generate ` which made those changes as Fix/Workaround . @sam-nguyen7 if you can give more context on this please do so.

File: controller/customdomain_utils.go (hunk 4 above)
Comment (dustman9000) [reply controller/customdomain_utils.go:238]: Fair enough, the code is small enough to not worry about the replication at this point.

File: deploy/crds/managed.openshift.io_customdomains.yaml (hunk 3 above)
Comment (dustman9000) [reply deploy/crds/managed.openshift.io_customdomains.yaml:52]: That's odd, there might be a bug with that make target then.
